panoramapublic/test/src/org/labkey/test/tests/panoramapublic/PanoramaWebPublicSearchTest.java

```diff
@@ -0,0 +1,184 @@
+
+import java.util.Arrays;
+
+@Category({External.class, MacCossLabModules.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 5)
+public class PanoramaWebPublicSearchTest extends PanoramaPublicBaseTest
+{
+    private static final String SKY_FILE_1 = "Study9S_Site52_v1.sky.zip";
+    private static final String SKY_FILE_2 = "Olga_srm_course_heavy_light_w_maxquant_lib.sky.zip";
+
+    private static final String SUBFOLDER_1 = "First Subfolder";
+    private static final String SUBFOLDER_2 = "Second Subfolder";
+
+    private static final String AUTHOR_FIRST_NAME = "Jane";
+    private static final String AUTHOR_LAST_NAME = "Doe";
+
+    ApiPermissionsHelper permissionsHelper = new ApiPermissionsHelper(this);
+
+    @BeforeClass
+    public static void initialSetUp()
+    {
+        PanoramaWebPublicSearchTest init = (PanoramaWebPublicSearchTest) getCurrentTest();
+        init.initialSetupHelper();
+    }
+
+    @Override
+    public String getSampleDataFolder()
+    {
+        return "panoramapublic/";
+    }
+
+    private void initialSetupHelper()
+    {
+        goToProjectHome();
+        portalHelper.addBodyWebPart("Panorama Public Search");
+
+        setupSubfolder(getProjectName(), SUBFOLDER_1, FolderType.Experiment);
+        importData(SKY_FILE_1, 1);
+        createExperimentCompleteMetadata("Test experiment for search improvements");
+
+        setupSubfolder(getProjectName(), SUBFOLDER_2, FolderType.Experiment);
+        importData(SKY_FILE_2, 1);
+
+        _userHelper.createUser(SUBMITTER);
+        permissionsHelper.addMemberToRole(SUBMITTER, "Folder Administrator", PermissionsHelper.MemberType.user, getProjectName() + "/" + SUBFOLDER_2);
+        impersonate(SUBMITTER);
+        updateSubmitterAccountInfo(AUTHOR_FIRST_NAME, AUTHOR_LAST_NAME, SUBFOLDER_2);
+        createExperimentCompleteMetadata("Submitter Experiment");
+        stopImpersonating();
+        portalHelper.exitAdminMode();
+    }
+
+    /*
+        First Name and last name for the user needs to be set for Author in experiment to get populated.
+     */
+    private void updateSubmitterAccountInfo(String firstName, String lastName, String subfolder)
+    {
+        goToMyAccount();
+        clickButton("Edit");
+        setFormElement(Locator.name("quf_FirstName"), firstName);
+        setFormElement(Locator.name("quf_LastName"), lastName);
+        clickButton("Submit");
+        navigateToFolder(getProjectName(), subfolder);
+    }
+
+    @Test
+    public void testExperimentalSearch()
+    {
+        goToProjectHome();
+        PanoramaPublicSearchWebPart panoramaPublicSearch = new PanoramaPublicSearchWebPart(getDriver(), "Panorama Public Search");
+        DataRegionTable table = panoramaPublicSearch.setAuthors(AUTHOR_FIRST_NAME).search();
+        CustomizeView customizeView = table.openCustomizeGrid();
+        customizeView.addColumn("Authors");
+        customizeView.clickViewGrid();
+        table = panoramaPublicSearch.refreshedTable();
```

Comment:
`DataRegionTable.applyCustomView` will wait for the data region to update. (`clickViewGrid` should probably be deprecated at some point since it does not)
```suggestion
        customizeView.applyCustomView(0);
```

In general, I don't think the `refreshedTable` method should be necessary as long as you use methods that wait for the data region to update.

Reply:
Done